src/Utils/hooks/useTracking.ts

```diff
@@ -0,0 +1,24 @@
+import { SystemContext } from "Artsy"
+import * as AnalyticsSchema from "Artsy/Analytics/Schema"
+import { useContext, useEffect } from "react"
+import { TrackingProp } from "react-tracking"
+
+export function useTracking(
+  newTrackingInstance?: TrackingProp
+): {
+  tracking: TrackingProp
+  AnalyticsSchema: typeof AnalyticsSchema
+} {
+  const { tracking, setTracking } = useContext(SystemContext)
+
+  useEffect(() => {
+    if (newTrackingInstance) {
+      setTracking(newTrackingInstance)
```

Comment:
What if another subtree with a different base tracking prop is registered under the same system context provider? Won't that destroy the context for the parent/sibling trees?

Reply:
I think what I'd suggest as an alternative to this would be to create our own version of `track` which injects a new context value (not kept in the system context). Then we also don't need to faff with the `useTracking(props.tracking)`

Something along these lines (just a rough sketch):

```ts
import { track as _track } from "react-tracking"

const track = trackingProps => Component =>
  _track(trackingProps)(props => (
    <ProvideTrackingContext trackingProps={props.tracking}>
      <Component {...props} />
    </ProvideTrackingContext>
  ))

```

And then useTracking would hook into the provided tracking context if present.

Reply:
I like this suggestion. It would mean less steps to set up tracking in a component before we can actually call `track.trackEvent`, and as a consumer of `useTracking`, that makes me feel more confident that I'm not going to forget something.

Reply:
@ds300 - I like it 👍  That would solve a few issues quite elegantly.